pkg/porter/list.go

```diff
@@ -15,68 +17,101 @@ type ListOptions struct {
 	printer.PrintOptions
 }
 
-// CondensedClaim holds a subset of pertinent values to be listed from a claim.Claim
-type CondensedClaim struct {
+// DisplayInstallation holds a subset of pertinent values to be listed from a claim.Claim
+type DisplayInstallation struct {
 	Name     string
 	Created  time.Time
 	Modified time.Time
 	Action   string
 	Status   string
+
+	Claim   claim.Claim
+	Result  claim.Result
+	Outputs DisplayOutputs
 }
 
-type CondensedClaimList []CondensedClaim
+func NewDisplayInstallation(installation claim.Installation) (DisplayInstallation, error) {
+	installTime, err := installation.GetInstallationTimestamp()
+	if err != nil {
+		return DisplayInstallation{}, err
+	}
+	c, err := installation.GetLastClaim()
+	if err != nil {
+		return DisplayInstallation{}, err
+	}
 
-func (l CondensedClaimList) Len() int {
+	var status string
+	r, err := c.GetLastResult()
+	if err == nil {
+		status = r.Status
+	} else {
+		status = claim.StatusUnknown
+	}
+
+	return DisplayInstallation{
+		Name:     installation.Name,
+		Created:  installTime,
+		Modified: c.Created,
+		Action:   c.Action,
+		Status:   status,
+		Claim:    c,
+		Result:   r,
+	}, nil
+}
+
+type DisplayInstallations []DisplayInstallation
+
+func (l DisplayInstallations) Len() int {
 	return len(l)
 }
-func (l CondensedClaimList) Swap(i, j int) {
+
+func (l DisplayInstallations) Swap(i, j int) {
 	l[i], l[j] = l[j], l[i]
 }
-func (l CondensedClaimList) Less(i, j int) bool {
+
+func (l DisplayInstallations) Less(i, j int) bool {
 	return l[i].Modified.Before(l[j].Modified)
 }
 
 // ListInstances lists installed bundles by their claims.
 func (p *Porter) ListInstances(opts ListOptions) error {
-	claims, err := p.Claims.ReadAll()
+	installations, err := p.Claims.ReadAllInstallationStatus()
 	if err != nil {
 		return errors.Wrap(err, "could not list bundle instances")
 	}
 
-	var condensedClaims CondensedClaimList
-	for _, claim := range claims {
-		condensedClaim := CondensedClaim{
-			Name:     claim.Installation,
-			Created:  claim.Created,
-			Modified: claim.Modified,
-			Action:   claim.Result.Action,
-			Status:   claim.Result.Status,
+	var displayInstallations DisplayInstallations
+	for _, installation := range installations {
+		displayInstallation, err := NewDisplayInstallation(installation)
+		if err != nil {
+			// There isn't an installation for this result
```

Comment:
Should we add a debug line here if p.Debug is set?

Reply:
Looking at that more closely the real problem is for situations where we can't determine when the bundle was installed, maybe because they have only run custom actions but not install yet. Let me handle that better.